### PR TITLE
working on VVC packing

### PR DESCRIPTION
**Pull request by yangpengjjj**

### CODE REVIEW

**Comment by podborski** [comment]
Please don't close and re-open. Just keep this one open and add stuff to your branch [yangpengjjj:Prepare_VVC](https://github.com/yangpengjjj/isobmff/tree/Prepare_VVC)

**Comment by yangpengjjj** [comment]
I made a mistake before, do I need to do other operations now？

**Comment by podborski** [comment]
> I made a mistake before, do I need to do other operations now？

No, its fine. It's just every time you open a new PR people get an email about it.

File: IsoLib/libisomediafile/src/MP4Atoms.h [hunk 1]
```diff
@@ -128,9 +128,14 @@ enum
   ISOItemReferenceAtomType                     = MP4_FOUR_CHAR_CODE('i', 'r', 'e', 'f'),
   ISOVCConfigAtomType                          = MP4_FOUR_CHAR_CODE('a', 'v', 'c', 'C'),
   ISOHEVCConfigAtomType                        = MP4_FOUR_CHAR_CODE('h', 'v', 'c', 'C'),
+  ISOVVCConfigAtomType                         = MP4_FOUR_CHAR_CODE('v', 'v', 'c', 'C'),
+  ISOVVCNALUConfigAtomType                     = MP4_FOUR_CHAR_CODE('v', 'v', 'n', 'C'),
   ISOAVCSampleEntryAtomType                    = MP4_FOUR_CHAR_CODE('a', 'v', 'c', '1'),
   ISOHEVCSampleEntryAtomType                   = MP4_FOUR_CHAR_CODE('h', 'v', 'c', '1'),
   ISOLHEVCSampleEntryAtomType                  = MP4_FOUR_CHAR_CODE('h', 'v', 'c', '2'),
+  ISOVVCSampleEntryAtomType                    = MP4_FOUR_CHAR_CODE('v', 'v', 'c', '1'),
+  ISOVVCSampleEntryAtomType2                   = MP4_FOUR_CHAR_CODE('v', 'v', 'i', '1'),
```
Comment (podborski) [inline IsoLib/libisomediafile/src/MP4Atoms.h]: bad naming: we should use in-band / out-of-band instead.

File: IsoLib/libisomediafile/src/TrackGroupTypeAtom.c [hunk 2]
```diff
@@ -87,6 +87,9 @@ static MP4Err setGroupID(struct MP4TrackGroupTypeAtom *self, u32 track_group_id)
   MP4Err err;
   err = MP4NoErr;
 
+  /* if use 'alte' track group, The value of (flags & 1) shall be equal to 1 to indicate the
+   * uniqueness of track_group_id */
+  self->flags          = 1;
```
Comment (podborski) [inline IsoLib/libisomediafile/src/TrackGroupTypeAtom.c]: We should not hard-code `flags=1` in the main library like that.

**Review by podborski — APPROVED** [review]
Good starting point. Tested the following things:

- demuxing Nokia's test vector `vvc_basic_track.mp4`
- decoding demuxed bitstream with Fraunhofer HHI `vvdec`
- muxing the bitstream back using `vvc_muxer`
- demuxing it with gpac's `MP4Box`
- verify the bitstream decoding again with Fraunhofer HHI `vvdec`

All of the above worked.

**Pull request merged**